scripts/utilities/datarun.py

```diff
@@ -0,0 +1,360 @@
+            (default True)'''
+
+        # set self.rescal_root using the environment or rescal_root argument
+        # should be the parent directory of the ReSCAL src and scripts directories
+        self.rescal_root = find_rescal_root(rescal_root)
+        
+        # this directory should alreay exist, but it will be created if it doesn't
+        if not os.path.isabs(experiment_parent_directory):
+            self.experiment_parent_directory = os.path.join(self.rescal_root,
+                                                            experiment_parent_directory)
+        else:
+            self.experiment_parent_directory = os.fspath(experiment_parent_directory)
+
+        # deal with experiment_directory
+        if not os.path.isabs(experiment_directory):
+            self.experiment_directory = os.path.join(self.experiment_parent_directory,
+                                                     experiment_directory)
+        else:
+            self.experiment_directory = os.fspath(experiment_directory)
+            
+        # set up paths to other files/directories of interest
+        self.rescal_src_directory = os.path.join(self.rescal_root, 'src')
+        self.scripts_directory = os.path.join(self.rescal_root, 'scripts')
+        self.real_data_directory = os.path.join(self.scripts_directory, 'real_data')
+        self.rescal_executable = os.path.join(self.scripts_directory, 'rescal')
+        self.genesis_executable = os.path.join(self.scripts_directory, 'genesis')
+        self.ui_xml = os.path.join(self.rescal_src_directory, 'rescal-ui.xml')
+        self.par = os.path.join(self.experiment_directory, 'run.par')
+        self.meta_data = os.path.join(self.experiment_directory, 'meta_data')
+        
+        # parameters needs to contain directory paths for .par file generation to work
+        paths_to_add = {'rescallocation' : self.scripts_directory,
+                        'real_data_location' : self.real_data_directory}
+
+        self.parameters = {**parameters, **paths_to_add}
+        
+        # for dealing with the data from ReSCAL
+        self.keep_height_maps = keep_height_maps
+        self.keep_ffts = keep_ffts
+        self.height_maps = []
+        self.ffts = []
+        self.height_maps_path = os.path.join(self.experiment_directory, 'height_maps.npz')
+        self.ffts_path = os.path.join(self.experiment_directory, 'ffts.npz')
+
+
+
+    def check_paths(self):
+        '''Verify that the files and directories that should already exist 
+        actually do exist.'''
+
+        dirs =  [self.rescal_root, self.scripts_directory,
+                 self.real_data_directory, self.experiment_parent_directory,
+                 self.rescal_src_directory]
+        files = [self.rescal_executable, self.genesis_executable, self.ui_xml]
+        for d in dirs:
+            if not os.path.isdir(d):
+                sys.stderr.write('ERROR: directory ' + d + ' not found.\n')
+                sys.exit(0)
```

Comment:
`exit(0)`  indicates a successful termination. Use a non-zero exit code. -1 is common.

Reply:
Done